src/msw/window.cpp

```diff
@@ -4705,6 +4705,89 @@ wxWindowMSW::MSWOnMeasureItem(int id, WXMEASUREITEMSTRUCT *itemStruct)
     return false;
 }
 
+// ---------------------------------------------------------------------------
+// DPI
+// ---------------------------------------------------------------------------
+
+/*extern*/
+int wxGetSystemMetrics(int nIndex, wxWindow* win)
+{
+    wxTopLevelWindow* tlw = NULL;
+    if ( win )
+    {
+        tlw = wxDynamicCast(wxGetTopLevelParent(win), wxTopLevelWindow);
+    }
+    else
+    {
+        wxWindow* window = static_cast<wxApp*>(wxApp::GetInstance())->GetTopWindow();
+        if ( window )
+        {
+            tlw = wxDynamicCast(wxGetTopLevelParent(window), wxTopLevelWindow);
+        }
+    }
+
+    if ( tlw )
+    {
+        wxDynamicLibrary dllUser32;
+        if ( dllUser32.Load(wxS("User32.dll"), wxDL_VERBATIM | wxDL_QUIET) )
```

Comment:
I think we should use `wxLoadedDLL` here.

Also, this is a really minor nitpick, but why capitalize `User32`, we don't do it in any of the other places (including one in this file).